### DISCUSSION TITLE
Problema con dispositivi TS0001 _TZ3000_tqlv4ug4

### DISCUSSION THREAD

**Discussion by excalibur470** [indomus/forum#1979] [Hardware & Firmware]
Ciao a tutti,
una domanda da un "niubbo", quindi perdonate le inesattezze.
Ho acquistato, a più riprese dei dispositivi TS0001 _TZ3000_tqlv4ug4.
Avendo riscontrato che lo switch esterno normalmente funziona come monostabile (Premi->On, Rilascio->Off), mentre volevo utilizzarli con dei normali pulsanti in modalità bistabile (Premi->On, Premi->Off), occorreva modificare la configurazione in Gestione Parametri ZigBee di Home Assistant, attraverso queste funzioni:
![image](https://user-images.githubusercontent.com/6024569/194839551-31f24301-fc8b-46fc-9a62-f190219b6d6c.png)
Il problema è che in altri apparati queste funzioni con ci siano, sebbene siano sempre dei TS0001 _TZ3000_tqlv4ug4.
Le differenze riguardano il quirk e la classe. Quelli che hanno la gestione riportano:
![image](https://user-images.githubusercontent.com/6024569/194839956-71bf400c-d651-4c1f-946e-6943cd646536.png)
e nella firma
![image](https://user-images.githubusercontent.com/6024569/194840354-424b60cd-a405-442a-bb0e-99e37972f05d.png)

Mentre gli altri:
![image](https://user-images.githubusercontent.com/6024569/194840036-f559b4e2-be6e-401f-839f-f89c945fec9f.png)
![image](https://user-images.githubusercontent.com/6024569/194840776-9f5e314e-d1ca-4d55-9f5f-b9cf6ce16b61.png)
Secondo voi c'è un modo per riprogrammarli, in modo che operino come gli altri, o sarò costretto a sostituirli?
Scusate la lunghezza e grazie.

**Comment by snakuzzo** [comment]
No...li devi cambiare, a meno che non sia previsto un aggiornamento del firmware del device da parte del produttore.
Oppure puoi provare a vedere se è una questione legata al supporto previsto da ZHA (stai usando quello, vero ?).
Controlla pure su zigbee2mqtt se ci sono differenze a livello software.

**Reply by excalibur470** [reply]
Brutta faccenda, perché non sono distinguibili gli uni dagli altri. Comunque grazie anche per la risposta fulminea.
